packages/notifications/push/amplify_push_notifications/lib/amplify_push_notifications.dart

```diff
@@ -110,51 +117,81 @@ class AmplifyPushNotifications extends PushNotificationsPluginInterface {
   Stream<PushNotificationMessage> get onNotificationOpened =>
       _onNotificationOpened;
 
-  @override
-  void onNotificationReceivedInBackground(OnRemoteMessageCallback callback) {
-    throw UnimplementedError(
-      'onNotificationReceivedInBackground() has not been implemented.',
-    );
-  }
-
   @override
   Future<void> configure({
     AmplifyConfig? config,
     required AmplifyAuthProviderRepository authProviderRepo,
   }) async {
-    // Parse config values from amplifyconfiguration.json
+    if (_isConfigured) {
+      return;
+    }
+    // TODO Parse config values from amplifyconfiguration.json for notifications
     final analyticsConfig = config?.analytics?.awsPlugin;
     if (analyticsConfig == null) {
       throw const AnalyticsException('No Pinpoint plugin config available');
     }
 
-    if (_isConfigured) {
-      return;
-    }
+    // Register the callback dispatcher
+    await _registerCallbackDispatcher();
 
     // Initialize Endpoint Client
-    await _serviceProviderClient.init(
-      config: config,
-      authProviderRepo: authProviderRepo,
-    );
-
-    // Block configure if registering device is not complete
-    // TODO: Catch and throw exceptions originating from token or device registration
-    final deviceToken = await onTokenReceived.first;
-    await _registerDevice(deviceToken);
+    await _initiliazeServiceClientAndRegisterDevice(config, authProviderRepo);
 
     // Initialize listeners
     onTokenReceived.listen(_tokenReceivedListener);
     onNotificationReceivedInForeground.listen(_foregroundNotificationListener);
     onNotificationOpened.listen(_notificationOpenedListener);
-
-    // TODO: Register the callback dispatcher
-
+    await _registerBgCallback(
```

Comment:
I don't follow the need for all these callbacks - why not just one?

Reply:
So we can record Analytics but that can be done in the callback dispatcher too before calling the external callback function. Do we want to change to that?

Reply:
I think that would make more sense, yeah